src/thx/Nel.hx

```diff
@@ -44,23 +61,100 @@ abstract Nel<A> (NonEmptyList<A>) from NonEmptyList<A> to NonEmptyList<A> {
       case ConsNel(x, xs): f(x, xs.fold(f));
     };
 
-  @:op(N+N0) 
-  public function append(nel: NonEmptyList<A>): Nel<A> 
+  @:op(N+N0)
+  public function append(nel: NonEmptyList<A>): Nel<A>
     // TODO: Implement imperatively for stack safety.
     return switch this {
       case Single(x): ConsNel(x, nel);
       case ConsNel(x, xs): ConsNel(x, xs.append(nel));
     };
 
+/**
+Gets the head item of the non-empty list, which is guaranteed to exist
+**/
+  public function head() : A {
+    return switch this {
+      case Single(x) : x;
+      case ConsNel(x, xs) : x;
+    };
+  }
+
+/**
+Gets the tail (all but the first element) of the `Nel<A>` as an `Option<Nel<A>>`
+**/
+  public function tail() : Option<Nel<A>> {
```

Comment:
It might make more sense to return just an ordinary List as the result, rather than Option<Nel<A>> ?